### PR TITLE
Adding objective function logger callback

### PR DESCRIPTION
**Pull request by g-poveda**
-new callback that logs current best objective function -example of use in one example script

### CODE REVIEW

File: discrete_optimization/generic_tools/callbacks/loggers.py [hunk 1]
```diff
@@ -49,3 +49,40 @@ def on_step_end(
         logger.log(
             msg=f"Iteration #{self.nb_iteration}", level=self.step_verbosity_level
         )
+
+
+class ObjectiveLogger(Callback):
+    """
```
Comment (nhuet) [inline discrete_optimization/generic_tools/callbacks/loggers.py:55]: Maybe the docstrings should be changed ;p

**Review by nhuet — CHANGES_REQUESTED** [review]

File: discrete_optimization/generic_tools/callbacks/loggers.py (hunk 1 above)
Comment (g-poveda) [reply discrete_optimization/generic_tools/callbacks/loggers.py:55]: well seen, actually same for the already existing NbIterationLogger right ?!

Comment (nhuet) [reply discrete_optimization/generic_tools/callbacks/loggers.py:55]: ... correct.

Comment (g-poveda) [reply discrete_optimization/generic_tools/callbacks/loggers.py:54]: Done

**Pull request merged**